modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -841,16 +947,25 @@ sub apply_tweaks {
                             'submission_cmd_args'   => $new_submission_cmd_args,
                             'worker_cmd_args'       => $new_worker_cmd_args,
                         );
+                        $tweakStructure->{Return}->{OldValue} = undef;
                     }
+                    $tweakStructure->{Return}->{Field} = "resource_class[$rc_name].meadow";
+                    $tweakStructure->{Return}->{NewValue} = stringify([$new_submission_cmd_args, $new_worker_cmd_args]);
                     $need_write = 1;
                 }
             }
+            push @{$responseStructure->{Tweaks}}, $tweakStructure;
```

Comment:
`$tweakStructure` needs to be pushed at every iteration of the for loop on `@resource_classes`, otherwise we only get the last tweak